utils/transformData.js

```diff
@@ -0,0 +1,87 @@
+const getDateFormat = (date) => {
+  const stringDate = `${date.getDate()}. ${
+    date.getMonth() + 1
+  }. ${date.getFullYear()}`;
+
+  return stringDate;
+};
+
+const dateOfPublicationFormat = (data) =>
+  data.map((item) => {
+    if (item.dateOfPublication) {
+      const transformedDate = getDateFormat(
+        new Date(item.dateOfPublication.toDate()),
+      );
+
+      return { ...item, dateOfPublication: transformedDate };
+    }
+
+    return item;
+  });
+
+export const transformSnapshot = (snapshot) => {
+  const transformedData = snapshot.docs.map((doc) => ({
+    ...doc.data(),
+    id: doc.id,
+  }));
+
+  return dateOfPublicationFormat(transformedData);
+};
+
+export const groupByFirstChar = (array, groupProperty) =>
+  array.reduce((acc, item) => {
+    const firstChar = item[groupProperty].charAt(0);
+
+    if (!acc[firstChar]) {
+      acc[firstChar] = []; // mutation
+    }
+
+    acc[firstChar].push(item); // even this mutates
+
+    return acc;
+  }, {});
+
+export const createSlugLink = (title) =>
+  title
+    .normalize('NFD')
+    .replace(/[\u0300-\u036f]/g, '')
+    .replace(/\s+/g, '-')
+    .toLowerCase();
+
+export const parseKeywords = (keywordsString) => {
+  if (keywordsString) {
+    const arrayKeywords = keywordsString.split(',');
+
+    arrayKeywords.forEach((item, index) => {
+      if (item.length < 2) {
+        arrayKeywords.splice(index, 1);
```

Comment:
Be careful about the big difference between slice and splice. Splice mutates the original array. Which is a big no-no, especially in a code like this.

Be super careful about mutation overall. It can lead to veeery unexpected behaviour, beyond the function.

If you want to remove items - use `someArray.filter(...)`. It will even shorten the code, no mutation, pure perfection